engine/src/main/java/org/hibernate/validator/HibernateValidatorConfiguration.java

```diff
@@ -186,4 +187,19 @@
 	 * @since 5.2
 	 */
 	HibernateValidatorConfiguration timeProvider(TimeProvider timeProvider);
+
+	/**
+	 * Registers an expression factory for use when constraint violation messages use EL.
+	 * This allows applications to avoid issues with the method {@link ExpressionFactory#newInstance()},
+	 * which uses the TCCL in some implementations.
+	 * @param expressionFactory The expression factory for use.
+	 * @return  {@code this} following the chaining method pattern
+	 * @since 5.3
+     */
+	HibernateValidatorConfiguration expressionFactory(ExpressionFactory expressionFactory);
```

Comment:
The only alternative I know is to wrap a setting of the TCCL to the 'externalClassLoader'  around the call to `ExpressionFactory.newInstance()` in `ELTermResolver`. The _defined API_ of is to read the TCCL, so it seemed to me that it was not implementation-dependent to do that -- presumably that behavior is sitting in a JSR somewhere.

However, @gunnarmorling was not enamored of that approach, which is why you are reading what you're reading.

I don't think I have any other paths to offer you. Also note that calling setContextClassLoader annoys the security manager in the TCK. I did code it that way at first and hit that.